planner_cspace/src/dummy_robot.cpp

```diff
@@ -55,9 +57,24 @@ class DummyRobot
   }
   void cbInit(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr &msg)
   {
-    x_ = msg->pose.pose.position.x;
-    y_ = msg->pose.pose.position.y;
-    yaw_ = tf::getYaw(msg->pose.pose.orientation);
+    tf::StampedTransform trans_msg_odom;
+    try
+    {
+      tfl_.lookupTransform(msg->header.frame_id, "odom", ros::Time(0), trans_msg_odom);
+    }
+    catch (tf::TransformException &e)
+    {
+      ROS_WARN("%s", e.what());
+      return;
+    }
+
+    tf::Pose pose;
+    tf::poseMsgToTF(msg->pose.pose, pose);
+    const tf::Transform trans_odom_baselink = trans_msg_odom.inverse() * pose;
```

Comment:
you can use `tf::TransformListener::transformPose` like https://github.com/at-wat/mcl_3dl/blob/master/src/mcl_3dl.cpp#L455-L463